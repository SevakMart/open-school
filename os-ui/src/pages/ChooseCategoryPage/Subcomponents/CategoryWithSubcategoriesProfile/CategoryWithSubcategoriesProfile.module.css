.mainContent h3 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 29px;
  color: #222427;
}/*# sourceMappingURL=CategoryWithSubcategoriesProfile.module.css.map */