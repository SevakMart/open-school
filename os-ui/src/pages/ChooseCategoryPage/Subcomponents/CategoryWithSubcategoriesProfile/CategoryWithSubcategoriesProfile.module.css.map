{"version":3,"sources":["CategoryWithSubcategoriesProfile.module.scss","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss","CategoryWithSubcategoriesProfile.module.css"],"names":[],"mappings":"AAKG;ECJD,yBCDY;EDEZ,kBDImC;ECHnC,gBDG0C;ECF1C,eDE8C;ECD9C,iBDCmD;EACjD,cGJgB;ACIpB","file":"CategoryWithSubcategoriesProfile.module.css"}