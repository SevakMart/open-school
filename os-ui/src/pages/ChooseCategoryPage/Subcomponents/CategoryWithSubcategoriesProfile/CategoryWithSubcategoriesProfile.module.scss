@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";

.mainContent {
    @include flexStyle(column,flex-start,none);
    width:25%;
    margin-bottom:5%;
    h3 {
        @include fontStyle('Montserrat',normal,700,1.5rem,1.813rem);
        color:#222427;
        padding-left:2%;
    }
    .subMainContent {
        @include flexStyle(column,flex-start,space-between);
    }
    .subcategoryContent{
        margin-bottom:3%;
        input{
            background-color:#D9DBE9;
            border-radius: 0.25rem;
        }
        label{
            @include fontStyle('Montserrat',normal,600,1.25rem,1.5rem);
            color:#5E617B;
        }
        margin-bottom:4%;
    }
}