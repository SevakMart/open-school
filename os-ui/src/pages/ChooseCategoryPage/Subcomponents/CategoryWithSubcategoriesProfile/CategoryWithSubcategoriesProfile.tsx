import { CategoryWithSubcategoriesProfilePropTypes } from '../../../../types/CategoryWithSubcategoriesProfilePropTypes';
import SubcategoryList from '../SubcategoryList/SubcategoryList';
import styles from './CategoryWithSubcategoriesProfile.module.scss';

/* eslint-disable max-len */

const CategoryWithSubcategoriesProfile = ({ parentCategory, subcategories }:CategoryWithSubcategoriesProfilePropTypes) => {
  const { mainContent } = styles;

  return (
    <div className={mainContent}>
      <h3 data-testid={parentCategory}>{parentCategory}</h3>
      <SubcategoryList subcategories={subcategories} />
    </div>
  );
};
export default CategoryWithSubcategoriesProfile;
