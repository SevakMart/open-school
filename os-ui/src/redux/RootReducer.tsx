import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
