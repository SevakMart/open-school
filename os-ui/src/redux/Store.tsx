import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
