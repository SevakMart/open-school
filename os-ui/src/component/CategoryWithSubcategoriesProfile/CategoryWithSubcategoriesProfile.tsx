import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../redux/Store';
import { addSubcategory, removeSubcategory } from '../../redux/Slices/ChoosSubcategorySlice';
import { CategoryWithSubcategoriesProfilePropTypes } from '../../types/CategoryWithSubcategoriesProfilePropTypes';
import styles from './CategoryWithSubcategoriesProfile.module.scss';

const CategoryWithSubcategoriesProfile = ({
  parentCategory, subcategories,
}:
    CategoryWithSubcategoriesProfilePropTypes) => {
  const subcategoryIdArray = useSelector<RootState>((state) => state.chooseSubcategories);
  const dispatch = useDispatch();
  const { mainContent, subcategoryContent } = styles;
  const handleChange = (e:React.SyntheticEvent) => {
    if ((e.target as HTMLInputElement).checked) {
      dispatch(addSubcategory(+(e.target as HTMLInputElement).id));
    } else dispatch(removeSubcategory(+(e.target as HTMLInputElement).id));
  };

  return (
    <div className={mainContent}>
      <h3>{parentCategory}</h3>
      {
            subcategories.length > 0 ? subcategories.map((subcategory) => (
              <div className={subcategoryContent} key={+subcategory.id}>
                {
                  (subcategoryIdArray as Array<number>).some((id) => id === +subcategory.id)
                    ? <input type="checkbox" id={`${subcategory.id}`} onChange={handleChange} checked />
                    : <input type="checkbox" id={`${subcategory.id}`} onChange={handleChange} checked={false} />
                }
                <label htmlFor={`${subcategory.id}`}>{subcategory.title}</label>
              </div>
            )) : null
        }
    </div>
  );
};
export default CategoryWithSubcategoriesProfile;
