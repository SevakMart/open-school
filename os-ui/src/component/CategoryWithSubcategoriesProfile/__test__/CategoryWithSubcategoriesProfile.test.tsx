import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PayloadAction } from '@reduxjs/toolkit';
import CategoryWithSubcategoriesProfile from '../CategoryWithSubcategoriesProfile';
import { store } from '../../../redux/Store';

const subcategoriesArray = [
  {
    id: 1,
    title: 'Arrays',
  },
  {
    id: 2,
    title: 'Functions',
  },
  {
    id: 3,
    title: 'Classes',
  },

];

jest.mock('redux-state-sync', () => ({
  createStateSyncMiddleware:
    () => () => (next: (action: PayloadAction) => void) => (action: PayloadAction) => next(action),
  initMessageListener: () => jest.fn(),
}));

describe('Create tests for the Component', () => {
  test('Create a Snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CategoryWithSubcategoriesProfile
          parentCategory="Java"
          subcategories={subcategoriesArray}
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Check the content of the component', () => {
    render(
      <Provider store={store}>
        <CategoryWithSubcategoriesProfile
          parentCategory="Java"
          subcategories={subcategoriesArray}
        />
      </Provider>,
    );
    const headerElement = screen.queryByTestId('Java');
    const labelElement = screen.queryByTestId('Functions');
    expect(headerElement).toBeInTheDocument();
    expect(headerElement).toHaveTextContent('Java');
    expect(labelElement).toBeInTheDocument();
    expect(labelElement).toHaveTextContent('Functions');
  });
});
