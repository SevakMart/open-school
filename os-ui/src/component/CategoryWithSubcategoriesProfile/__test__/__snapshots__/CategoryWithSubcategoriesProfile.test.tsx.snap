// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create tests for the Component Create a Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContent"
  >
    <h3
      data-testid="parentCategoryHeader"
    >
      Java
    </h3>
    <div
      class="subcategoryContent"
    >
      <input
        id="1"
        type="checkbox"
      />
      <label
        data-testid="Arrays"
        for="1"
      >
        Arrays
      </label>
    </div>
    <div
      class="subcategoryContent"
    >
      <input
        id="2"
        type="checkbox"
      />
      <label
        data-testid="Functions"
        for="2"
      >
        Functions
      </label>
    </div>
    <div
      class="subcategoryContent"
    >
      <input
        id="3"
        type="checkbox"
      />
      <label
        data-testid="Classes"
        for="3"
      >
        Classes
      </label>
    </div>
  </div>
</DocumentFragment>
`;
